version-bump-pr: merge main with '-X theirs' to handle first-release CHANGELOG conflicts

## Context

The `version-bump-pr` composite action runs after a successful
release to open a PR that back-merges `main` into `develop` and
bumps to the next patch version. The "back-merge main" step today
is:

```yaml
- name: Create bump branch and merge main
  if: steps.check.outputs.needed == 'true'
  shell: bash
  run: |
    git checkout -b "${{ steps.next.outputs.branch }}" origin/develop
    git merge origin/main --no-edit
```

No merge strategy flag.

## Problem

On any **first-ever release** of a repo, `develop` and `main` have
diverged on files the release pipeline writes (notably `CHANGELOG.md`
and anything under `releases/`). Before the first release:

- `develop` has a scaffolding CHANGELOG (header only) or a pre-
  existing one from an earlier aborted release attempt
- `main` has the generated CHANGELOG from the just-completed
  release plus `releases/v<version>.md`

`git merge origin/main --no-edit` with no strategy flag produces an
add/add conflict on CHANGELOG.md, the step exits 1, and the entire
auto-bump-PR step fails.

Hit live on standard-tooling-plugin v1.3.0 on 2026-04-23:

```text
Auto-merging CHANGELOG.md
CONFLICT (add/add): Merge conflict in CHANGELOG.md
Automatic merge failed; fix conflicts and then commit the result.
##[error]Process completed with exit code 1.
```

mq-rest-admin-* repos don't hit this — they've been releasing long
enough that develop and main stay in sync between cycles (each bump
PR brings main's CHANGELOG into develop, so the NEXT bump-merge is
a no-op on that file). First release is the pain.

## Fix

Change the merge to prefer main's content on conflict:

```yaml
- name: Create bump branch and merge main
  if: steps.check.outputs.needed == 'true'
  shell: bash
  run: |
    git checkout -b "${{ steps.next.outputs.branch }}" origin/develop
    git merge origin/main --no-edit -X theirs
```

Why `-X theirs` is correct: the intent of the bump PR is "develop
now reflects the post-release state of main." Any file where main
and develop diverge at this point should take main's version —
main has the tag, the CHANGELOG, and whatever else the release
pipeline wrote. `-X ours` (which st-prepare-release uses for the
main-into-release merge) would be wrong here; we'd WANT main's
post-release state.

No downsides observed: the bump branch is ephemeral and exists
solely to carry main's state back into develop plus the version
bump.

## Also relevant

- wphillipmoore/standard-tooling-plugin#63 — the manual 1.3.1 bump
  PR that was needed because the automated version-bump-pr step
  failed. Used `-X theirs` manually; clean merge.
- wphillipmoore/standard-tooling#268 — sibling fix for
  `st-submit-pr`'s auto-merge call. Different tool, different
  failure mode; both belong in the "first-real-run gotchas we hit
  on 2026-04-23" bucket.

## Acceptance criteria

- [ ] `actions/publish/version-bump-pr/action.yml` merge step
  passes `-X theirs`
- [ ] Comment in the action explains the rationale (post-release
  back-merge intent)
- [ ] First-release scenario verified: a fresh repo with only a
  scaffolding CHANGELOG produces a clean bump PR after first
  release

## Priority

Moderate. Existing releasing repos (mq-rest-admin-*) aren't
affected. Bites at first-release time, which matters when
onboarding new repos onto the publish pipeline.

--- a/actions/docs-deploy/action.yml
+++ b/actions/docs-deploy/action.yml
@@ -68,13 +68,18 @@ runs:
 
     - name: Configure git identity
       shell: bash
+      # --global avoids depending on cwd being inside the repo. Composite
+      # shell steps in container jobs do not reliably land in
+      # $GITHUB_WORKSPACE when setup-python is skipped (pre-installed mike),
+      # so repo-scoped git config would fail with "not in a git directory".
       run: |
-        git config user.name "github-actions[bot]"
-        git config user.email "github-actions[bot]@users.noreply.github.com"
+        git config --global user.name "github-actions[bot]"
+        git config --global user.email "github-actions[bot]@users.noreply.github.com"
 
     - name: Determine version
       id: version
       shell: bash
+      working-directory: ${{ github.workspace }}
       run: |
         if [ "${{ github.ref }}" = "refs/heads/main" ]; then
           VERSION=$(${{ inputs.version-command }})
@@ -87,6 +92,7 @@ runs:
 
     - name: Stage changelog and release notes
       shell: bash
+      working-directory: ${{ github.workspace }}
       run: |
         DOCS_DIR=$(dirname "${{ inputs.mkdocs-config }}")/docs
 
@@ -110,6 +116,7 @@ runs:
 
     - name: Patch mkdocs nav with release versions
       shell: bash
+      working-directory: ${{ github.workspace }}
       run: |
         DOCS_DIR=$(dirname "${{ inputs.mkdocs-config }}")/docs
 
@@ -123,6 +130,7 @@ runs:
 
     - name: Deploy docs
       shell: bash
+      working-directory: ${{ github.workspace }}
       run: |
         if [ -n "${{ steps.version.outputs.alias }}" ]; then
           ${{ inputs.mike-command }} deploy -F ${{ inputs.mkdocs-config }} --push --update-aliases ${{ steps.version.outputs.version }} ${{ steps.version.outputs.alias }}

--- a/actions/publish/version-bump-pr/action.yml
+++ b/actions/publish/version-bump-pr/action.yml
@@ -108,9 +108,13 @@ runs:
     - name: Create bump branch and merge main
       if: steps.check.outputs.needed == 'true'
       shell: bash
+      # -X theirs resolves add/add CHANGELOG conflicts on first-release repos
+      # where develop's scaffolding CHANGELOG diverges from main's generated
+      # one. Intent of this back-merge is "develop reflects post-release main",
+      # so taking main's side on conflict is always correct here.
       run: |
         git checkout -b "${{ steps.next.outputs.branch }}" origin/develop
-        git merge origin/main --no-edit
+        git merge origin/main --no-edit -X theirs
 
     - name: Update version file
       if: steps.check.outputs.needed == 'true'

